File: lib/src/common/position.dart
class GeckoPosition {
  int x;
  int y;

  GeckoPosition(this.x, this.y);

  GeckoPosition.fromMap(Map<Object?, Object?> map)
      : x = map["x"] as int,
        y = map["y"] as int;

  Map<String, dynamic> toMap() => {
    "x": x,
    "y": y
  };
}

class GeckoOffset {
  int x;
  int y;

  GeckoOffset(this.x, this.y);

  GeckoOffset.fromMap(Map<Object?, Object?> map)
      : x = map["x"] as int,
        y = map["y"] as int;

  Map<String, dynamic> toMap() => {
    "x": x,
    "y": y
  };
}

File: android/src/main/kotlin/info/xert/gecko_view_flutter/common/Position.kt
package info.xert.gecko_view_flutter.common

data class Position(val x: Int, val y: Int): OutputStructure {
    override fun toMap(): Map<String, Any?> {
        val map = mutableMapOf<String, Any?>()
        map["x"] = x
        map["y"] = y
        return map
    }

    companion object: InputStructure<Position> {
        override fun fromMap(inputMap: Map<*, *>): Position {
            return Position(
                    x = inputMap["x"] as Int,
                    y = inputMap["y"] as Int
            )
        }
    }
}

data class Offset(val x: Int, val y: Int): OutputStructure {
    override fun toMap(): Map<String, Any?> {
        val map = mutableMapOf<String, Any?>()
        map["x"] = x
        map["y"] = y
        return map
    }

    companion object: InputStructure<Offset> {
        override fun fromMap(inputMap: Map<*, *>): Offset {
            return Offset(
                    x = inputMap["x"] as Int,
                    y = inputMap["y"] as Int
            )
        }
    }
}